### PR TITLE
[Merged by Bors] - chore(SetTheory/Cardinal/Basic): generalize `bddAbove_range`

### PR DESCRIPTION
**Pull request by vihdzp**
In truth we could simply replace all instances of `bddAbove_range` with `bddAbove_of_small`, but I've kept the lemma anyways for discoverability.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/SetTheory/Cardinal/Basic.lean [hunk 1]
```diff
@@ -6,6 +6,7 @@ Authors: Johannes Hölzl, Mario Carneiro, Floris van Doorn
 import Mathlib.Data.Fintype.BigOperators
 import Mathlib.Data.Set.Countable
 import Mathlib.Logic.Small.Set
+import Mathlib.Logic.UnivLE
```
Comment (vihdzp) [inline Mathlib/SetTheory/Cardinal/Basic.lean:9]: This import is required for the `Small.{max u v, u}` instance. This should be documented somewhere...

File: Mathlib/SetTheory/Cardinal/Basic.lean [hunk 2]
```diff
@@ -842,18 +836,19 @@ instance (a : Cardinal.{u}) : Small.{u} (Set.Iio a) :=
...
+    simpa using le_sum (fun x ↦ e.symm x) (e ⟨a, ha⟩)⟩
 
 theorem bddAbove_of_small (s : Set Cardinal.{u}) [h : Small.{u} s] : BddAbove s :=
   bddAbove_iff_small.2 h
 
+@[deprecated bddAbove_of_small (since := "2024-09-22")]
+theorem bddAbove_range {ι : Type u} (f : ι → Cardinal.{max u v}) : BddAbove (Set.range f) :=
+  ⟨sum f, by
+    rintro a ⟨i, rfl⟩
+    exact le_sum f i⟩
```
Comment (eric-wieser) [inline Mathlib/SetTheory/Cardinal/Basic.lean:850]: Can you prove this using `bddAbove_of_small` to prove it really is a replacement?

Comment (eric-wieser) [reply Mathlib/SetTheory/Cardinal/Basic.lean:850]: My hunch would be that actually it's better to keep this lemma around for discoverability.

Comment (vihdzp) [reply Mathlib/SetTheory/Cardinal/Basic.lean:850]: Using this lemma is almost always strictly less convenient. It usually requires explicit universe annotations. And it won't work for functions `a -> b` where `b : Type u` and `a : Small.{u} b`.

Comment (vihdzp) [reply Mathlib/SetTheory/Cardinal/Basic.lean]: In fact, my original plan was to do this universe generalization (#16394), until I realized `bddAbove_of_small` already subsumed it.

I feel like keeping a weaker version of another theorem might cause confusion later down the line, e.g. someone might wonder why `bddAbove_of_range` has restricted universes, or why other theorems like `bddAbove_iUnion` don't exist, etc.

**Review by YaelDillies — COMMENTED** [review]
Since you can't similarly delete `bddAbove_image`, are you sure it is not better for discoverability to keep `bddAbove_range` around but generalised to functions with small domain?

**Comment by vihdzp** [comment]
Eh, fine. I'll un-deprecate it.

File: Mathlib/Data/W/Cardinal.lean [hunk 3]
```diff
@@ -73,7 +73,7 @@ theorem cardinal_mk_le_max_aleph0_of_finite' [∀ a, Finite (β a)] :
                 (by
                   rw [succ_zero]
                   obtain ⟨a⟩ : Nonempty α := hn
-                  refine le_trans ?_ (le_ciSup (bddAbove_range.{_, v} _) a)
+                  refine le_trans ?_ (le_ciSup (bddAbove_of_small _) a)
```
Comment (YaelDillies) [inline Mathlib/Data/W/Cardinal.lean:76]: Can you restore?

**Review by YaelDillies — APPROVED** [review]
maintainer merge

once you have restored ;)

File: Mathlib/SetTheory/Cardinal/Basic.lean (hunk 1 above)
Comment (eric-wieser) [reply Mathlib/SetTheory/Cardinal/Basic.lean:11]: Documented how? Like any import, presumably removing it breaks the file.

Comment (vihdzp) [reply Mathlib/SetTheory/Cardinal/Basic.lean:11]: Well, the module docstring for `Logic.UnivLE` says nothing about how the `UnivLE` API indirectly provides this instance. I'll make a PR to add this statement.

**Comment by urkud** [comment]
Thanks! :tada:
bors merge

**Pull request closed without merging**